packages/nodejs/src/__tests__/transmitter.test.ts

```diff
@@ -0,0 +1,168 @@
+import { Transmitter } from "../transmitter"
+import nock from "nock"
+import https from "https"
+import fs from "fs"
+import { ClientRequest, IncomingMessage } from "http"
+
+describe("Transmitter", () => {
+  let originalEnv: any = undefined
+
+  beforeEach(() => {
+    if (!nock.isActive()) {
+      nock.activate()
+    }
+    nock.disableNetConnect()
+
+    originalEnv = { ...process.env }
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+    nock.restore()
+
+    process.env = originalEnv
+  })
+
+  describe(".transmit", () => {
+    function sampleTransmitter(): Transmitter {
+      return new Transmitter("https://example.com/foo", "request body")
+    }
+
+    function mockSampleRequest({
+      query = {},
+      responseBody = { json: "response" }
+    }: any = {}) {
+      nock("https://example.com")
+        .post("/foo", "request body")
+        .query({
+          api_key: "",
+          name: "",
+          environment: "test",
+          hostname: "",
+          ...query
+        })
+        .reply(200, responseBody)
+    }
+
+    async function expectSampleResponse(
+      response: Promise<any>,
+      expectedBody: object = { json: "response" }
+    ) {
+      await expect(response).resolves.toEqual({
+        status: 200,
+        body: expectedBody
+      })
+    }
+
+    function mockHTTPSRequest() {
+      return jest.spyOn(https, "request").mockImplementation(opts => {
+        const clientRequestMock = {
+          on: () => {},
+          write: () => {},
+          end: () => {}
+        }
+
+        return (clientRequestMock as unknown) as ClientRequest
+      })
+    }
+
+    it("resolves to a response on success", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest()
+
+      await expectSampleResponse(transmitter.transmit())
+    })
+
+    it("resolves to an empty object if the response is not JSON", async () => {
+      const transmitter = sampleTransmitter()
+
+      mockSampleRequest({ responseBody: "not JSON" })
+
+      await expectSampleResponse(transmitter.transmit(), {})
+    })
+
+    it("rejects on error", async () => {
+      const transmitter = sampleTransmitter()
+
+      // do not mock the request -- beforeAll uses nock to disable network
+      // requests, so any request to a non-mocked host will throw an error
+
+      await expect(transmitter.transmit()).rejects.toMatchObject({
+        error: expect.any(Error)
+      })
+    })
+
+    it("uses config values in the query string", async () => {
+      process.env.APPSIGNAL_PUSH_API_KEY = "some_api_key"
+      process.env.APPSIGNAL_APP_NAME = "myApp"
+      ;(process.env as any).NODE_ENV = "development"
```

Comment:
Is the semicolon needed here?

Reply:
Yup -- without it, that line and the previous one are interpreted as one, that is:
```typescript
... = "myApp"(process.env as any).NODE_ENV = ...
```

I had put semicolons at the end of all these lines, for consistency, but prettier has interesting ideas of what pretty means, so it removed the unnecessary ones, and moved the necessary one there instead. 🤷